dbt/adapters/bigquery/impl.py

```diff
@@ -678,20 +683,64 @@ def expand_target_column_types(cls, profile, project_cfg, temp_table,
         pass
 
     @classmethod
-    def _flat_columns_in_table(cls, profile, project_cfg, schema_name,
-                               table_name):
+    def _flat_columns_in_table(cls, table):
         """An iterator over the flattened columns for a given schema and table.
         Resolves child columns as having the name "parent.child".
         """
-        cols = cls.get_columns_in_table(profile, project_cfg,
-                                        schema_name, table_name)
-        for col in cols:
+        for col in cls.get_dbt_columns_from_bq_table(table):
             flattened = col.flatten()
             for subcol in flattened:
                 yield subcol
 
+    @classmethod
+    def _get_stats_column_names(cls):
+        """Construct a tuple of the column names for stats. Each stat has 4
+        columns of data.
+        """
+        columns = []
+        stats = ('num_bytes', 'num_rows', 'location', 'partitioning_type')
+        stat_components = ('label', 'value', 'description', 'include')
+        for stat_id in stats:
+            for stat_component in stat_components:
+                columns.append('stats:{}:{}'.format(stat_id, stat_component))
+        return tuple(columns)
+
+    @classmethod
+    def _get_stats_columns(cls, table):
+        """Given a table, return an iterator of key/value pairs for stats
+        column names/values.
+        """
+        column_names = cls._get_stats_column_names()
+        # cast num_bytes/num_rows to str before they get to agate, or else
+        # agate will incorrectly decide they are booleans.
+        column_values = (
+            'Number of bytes',
+            str(table.num_bytes),
+            'The number of bytes this table consumes',
+            True,
```

Comment:
Can we exclude `num_bytes`, `num_rows`, and `partitioning_type` if the relation is not a table? I think the code should be `relation.type == 'table'`. These fields aren't relevant for views or external relations, for instance

Reply:
I'm going to handle this by replacing setting `True` with `relation.type == 'table'` and then the downstream filtering will handle everything.

Reply:
that sounds like a good approach to me